### PR TITLE
Feat: onboarding vault result screen

### PR DESCRIPTION
**Pull request by Asmadek**
* New polkadot vault (with dynamic derivation) result screen
* use effector for wallet creation 

image

### CODE REVIEW

File: src/renderer/entities/wallet/ui/DerivedAccount/DerivedAccount.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        className,
+      )}
+    >
+      <Icon size={iconSize} name={KeyIcon[keyType]} className="shrink-0 text-text-secondary" />
+
+      <BodyText className="text-text-secondary">{t(name)}</BodyText>
+      <FootnoteText className="flex-1 text-text-tertiary" align="right">
+        {showDerivationPath ? derivationPath : ''}
+      </FootnoteText>
+    </div>
```
Comment (sokolova-an) [inline src/renderer/entities/wallet/ui/DerivedAccount/DerivedAccount.tsx:75]: Maybe have it as 
``` 
{showDerivationPath && 
  <FootnoteText className="flex-1 text-text-tertiary" align="right">
        {derivationPath}
      </FootnoteText>

**Review by sokolova-an — COMMENTED** [review]
Do we need both UseAltKeyPressed and useKeyPressed? And are they for the future, couldn't find where we use them?

**Comment by pgolovkin** [comment]
Feedback from Sabit:
1. Make opening of the spoiler with keys more smooth.
2. When I tap Option (Alt) make spoiler opening more smooth.

File: src/renderer/shared/core/types/account.ts [hunk 2]
```diff
@@ -23,7 +23,7 @@ export type BaseAccount = AbstractAccount & {
 // };
 
 export type ChainAccount = BaseAccount & {
-  baseId: BaseAccount['id'];
+  baseId?: BaseAccount['id'];
```
Comment (tuul-wq) [inline src/renderer/shared/core/types/account.ts:26]: This is a big change, why it became optional?

File: src/renderer/shared/ui/Icon/data/keyType.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import DDMainImg, { ReactComponent as DDMainSvg } from '@images/keyTypes/dd-main.svg';
+import DDHotImg, { ReactComponent as DDHotSvg } from '@images/keyTypes/dd-hot.svg';
+import DDCustomImg, { ReactComponent as DDCustomSvg } from '@images/keyTypes/dd-custom.svg';
+import DDGovernanceImg, { ReactComponent as DDGovernanceSvg } from '@images/keyTypes/dd-governance.svg';
+import DDPublicImg, { ReactComponent as DDPublicSvg } from '@images/keyTypes/dd-public.svg';
+import DDStakingImg, { ReactComponent as DDStakingSvg } from '@images/keyTypes/dd-staking.svg';
+
+const KeyTypeImaes = {
+  ddMain: { img: DDMainImg, svg: DDMainSvg },
```
Comment (tuul-wq) [inline src/renderer/shared/ui/Icon/data/keyType.tsx:9]: Maybe rename to `keyMain` and so on?
`const KeyTypeImages = {`

File: src/renderer/widgets/WalletDetails/lib/utils.ts [hunk 4]
```diff
@@ -5,14 +5,19 @@ export const wcDetailsUtils = {
...
 export const walletDetailsUtils = {
   isForgetModalOpen,
 };
 
 function isNotStarted(step: ReconnectStep, connected: boolean): boolean {
-  return step === ReconnectStep.NOT_STARTED && connected;
+  return [ReconnectStep.NOT_STARTED, ReconnectStep.CONFIRMATION].includes(step) && connected;
+}
+
+function isConfirmation(step: ReconnectStep): boolean {
```
Comment (tuul-wq) [inline src/renderer/widgets/WalletDetails/lib/utils.ts:19]: 👍

**Review by tuul-wq — CHANGES_REQUESTED** [review]
Nice 👍

File: src/renderer/entities/wallet/ui/DerivedAccount/DerivedAccount.tsx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+const KeyTitle: Record<KeyType, string> = {
+  [KeyType.CUSTOM]: 'derivedAccount.customTitle',
+  [KeyType.GOVERNANCE]: 'derivedAccount.governanceTitle',
+  [KeyType.HOT]: 'derivedAccount.hotTitle',
+  [KeyType.MAIN]: 'derivedAccount.mainTitle',
+  [KeyType.PUBLIC]: 'derivedAccount.publicTitle',
+  [KeyType.STAKING]: 'derivedAccount.stakingTitle',
+};
+
+type ShardedKeyType = KeyType.CUSTOM | KeyType.GOVERNANCE | KeyType.MAIN | KeyType.STAKING;
```
Comment (egor0798) [inline src/renderer/entities/wallet/ui/DerivedAccount/DerivedAccount.tsx:26]: Maybe move it to account.ts near KeyType? I think we can use it in multiple places

File: src/renderer/shared/lib/hooks/useKeyPressed.ts [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+ *
+ * @param {Key} targetKey - the key to check if it is being pressed
+ * @param {boolean} withMetaKey - (optional) whether to consider the meta key (e.g. Ctrl or Command) in the check
+ * @return {boolean} true if the key is being pressed, false otherwise
+ */
+export const useKeyPressed = (targetKey: Key, withMetaKey?: boolean): boolean => {
+  const [keyPressed, setKeyPressed] = useState<boolean>(false);
+
+  const downHandler = ({ key, metaKey, ctrlKey }: KeyboardEvent) => {
+    const isMetaPressed = withMetaKey ? metaKey || ctrlKey : true;
```
Comment (egor0798) [inline src/renderer/shared/lib/hooks/useKeyPressed.ts:16]: Are you sure abount Ctrl key here? Cause for windows Win key is a meta key

File: src/renderer/assets/images/keyTypes/dd-governance.svg
[the remark this answers is not in the document]
Comment (Asmadek) [reply src/renderer/assets/images/keyTypes/dd-governance.svg:1]: in the design we have such size

File: src/renderer/pages/Onboarding/Vault/ManageDynamicDerivations/model/manage-dynamic-derivations-model.ts [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+  source: {
+    walletForm: $walletForm.$values,
+    accounts: $accounts,
+  },
+  mapParams: (_, { walletForm, accounts }) => {
+    return {
+      wallet: {
+        name: walletForm.name.trim(),
+        type: WalletType.POLKADOT_VAULT,
+        signingType: SigningType.PARITY_SIGNER,
```
[the remark this answers is not in the document]
Comment (Asmadek) [reply src/renderer/pages/Onboarding/Vault/ManageDynamicDerivations/model/manage-dynamic-derivations-model.ts:118]: we don't have `SigningType.POLKADOT_VAULT` right now

File: src/renderer/shared/core/types/account.ts (hunk 2 above)
Comment (Asmadek) [reply src/renderer/shared/core/types/account.ts]: We need it for derived accounts, they don't have base account at all, only chain accounts

**Review by tuul-wq — APPROVED** [review]

**Pull request merged**